antismash/modules/nrps_pks/transat_ks_analysis/test/test_transat_ks_analysis.py

```diff
@@ -0,0 +1,40 @@
+# License: GNU Affero General Public License v3 or later
+# A copy of GNU AGPL v3 should have been included in this software package in LICENSE.txt.
+
+# for test files, silence irrelevant and noisy pylint warnings
+# pylint: disable=no-self-use,protected-access,missing-docstring
+
+import unittest
+
+from antismash.modules.nrps_pks.transat_ks_analysis import transat_ks_analysis
+
+
+class TestKScalls(unittest.TestCase):
+    query_ks = {'nrpspksdomains_ctg1_11_PKS_KS.1': 'IAIVGMGGIVPQAVDLDQFWDGLISGRSFIEELPKTHPEYDLYGKVFGDQKVWGGFIKEVDEFDASFFNISPREAEQMDPRQRLLLQLVWQTLEHAGHIPGDFSAGSMGVYIGMVGYSDYGELARYYNQNIEGHTLSGVSASIIANRISYFFDFKGPSITTDTACSSSLVGLLQAVSAIRSGQCEAALAGGINLMLSPFYHRAIQEIGILSEDGCCRTFDKAANGCVRGEGGGLLLLRPLDKALADHDTIYGIIKGSAENHGGTTNSLTAPSSEAQANLLEEAYRESMIPPSKVGYIEVHGTGTRLGDPAEINGLKEAFRKLHNQYDEPLSETVSCGLGAVKTNIGHLEPAAGVMGVMKVLLAMRHKQLPPTINFIEQNPYIEIENTPFYLVTRAQDWQSPKDDEGGMLARCAGVSSFGFGGANAHIVLEEY',
+                'nrpspksdomains_ctg1_11_PKS_KS.2': 'IAIIGLSGRYPEAENVNEFWDNLKTSKNCIREIPQERWDWKQYFDEEKGKAGKIYTKWGGFLKEVDKFDPLFFQISPRDAEQMDPQERLFLEEAYKSIEDSGYTPETLCKSRKIGVFVGVMNGTYAPQTRFFSMANRISYHLNFQGPSMAVDTACSASLTAIHLALESLYSGMSECAIAGGVNLILRPIHYMGLSAMTMLSSTNENRSFGDHADGFVDGEGVGAVVLKSLKAAIADDDHIYGILKGSVVNAGGKTNGYTVPNPNAQYELVLEALKRSGIDARAVSCIEAHGTGTVLGDPIEVSGLARAFEHYSKDRQFCTIGSVKSNIGHCESAAGIAGVTKVLLQMQQGQIVPSLHSKRLNPNIDFSNTPFVVQQELENWNRPVLKVDGQKKEYPRIAGISSFGAGGANAHVVIEEY',
+                'nrpspksdomains_ctg1_11_PKS_KS.3': 'IAIVGLSGRYPQAPDLAAYWQNLRNGKDCITEVPPDRWDWKKYYTEDRNQGGRHYSKWGGFIEDVDKFDPLFFNISPREAELIDPQERLFLEHVWMALEDAGYRGEDLQGEAGEYLGGQVGVYAGVMYGEYQLFAAEESVRGNPLSVGGSYASIANRISYVLNLHGPSMSIDTMCSSSLTALHLACQDLKLGRTNLGVAGGVNVTIHPNKYLMLSNGQFISSGGHCESFGKGGDGYIPGEGVGVALLKRLSDAQRDGDHIYGVIKGSGLNHGGKTHGYSVPNPKAQQTVISRALKESGIRPGAISYIEAHGTGTKLGDPIEITGLSKAFGSEVQRGSCFIGSAKSNIGHCESAAGIAGVTKVLLQMKYGQIVPSLHSQVLNPNIDFSQTPFVVQQELKEWFRPLLKLNGTRKEYPRIAGVSSFGAGGANAHLVIEEY',
+                'nrpspksdomains_ctg1_12_PKS_KS.1': 'IAIVGMAGKYPGAKDLEEYWKNLVGGKNSIQEVPLSRWDVNQYYDPAPSKPGKVNCKWLGLLDDIDCFDPLFFMISPAEAEEMDPQHRIFLEEGHKAFESAGYSSSGLSNKKCGVYLGIMSNEYSLLAIEAQRSMSGTSNNYAIGAARMAYYLNLKGPAIPIDTACSSSLVGTHLACQALHAGEIDMALVGGVTLYLTAESYIDMCASGMLSAEGQCKTFDDSADGFVPGEGVGAIVLKRLSEAQRNDDCILGTIIGSGINQDGKTNGITAPSVKSQIELEREIYRKYKIDPSSISYVETHGTGTKLGDPIELEALSTVFKEQTDRKNYCALGSVKSNIGHTSAAAGVASMHKALLSLKHQKLVPTLNFKKPNSHFNFDESPFYVNTELKRWESEGEKPRRVGVSSFGFSGTNAHVVIEEY',
+                'nrpspksdomains_ctg1_12_PKS_KS.2': 'IAIIGLSGRYPQAYNVEEYWNHLKTGKDCITEIPRDRWSLEGFFHEDFEEAVAQGKSYSKWGGFLEGFSEFDPLFFNISSHEAMSMDPQERLFLQTCWEVCEDGGYTREKIAIQHHGKVGVFAGITKTGFDLYGPELWKKGEPVFPHTSFSSVANRVSYLLNLKGPSMPIDTMCSSSLTAIHEACKHVDQGECELAIAGGVNLYLHPSTYRGLCAQRMLALDRQCRSFGKGGNGFVPGEGVGAVLLKRLSQALFDGDHIYGVIRGTSANHGGKTNGYTVSNPKVQAELIRETLDKTGIDARTVSYIEAHGTATELGDPIEVAGLSQAFQQDTVDTGFCALGSAKSNVGHLEAAAGMAGVTKILLQMKHKVLVPSLHAKELNPNINFSKTPFVVQQELGEWRSPFLERDGAGKEYPRIAGISSFGAGGANAHVIIEEY'}
+
+    def test_transpact(self):
+        results = run_transpact_ks_analysis(query_ks)
```

Comment:
This function is undefined at this point, so nothing is being tested at all.

Also, unit tests shouldn't be executing external programs, we have nearly 900 unit tests that run in less than 6 seconds on my laptop, but this single line adds one test that takes 25 seconds alone. Either mock the output of the external commands or move it to an integration test (by changing the filename from `test_*` to `integration_*` and running with `make integration` instead).